tests/test_scheduler/test_unit/test_base.py

```diff
@@ -0,0 +1,286 @@
+                                'uid': 2,
+                                'lfs': {'path': 'test', 'size': 100},
+                                'gpus':[0,1,0]}]
+
+    slots = {"cores_per_node": 16,
+             "lfs_per_node": {"size": 0, "path": "/dev/null"},
+             "nodes": [{"lfs": {"path": "/dev/null", "size": 0},
+                        "core_map": [[0]],
+                        "name": "a",
+                        "gpu_map": [[1]],
+                        "uid": 2,"mem": 10}],
+             "lm_info": "INFO",
+             "gpus_per_node": 6,
+             }
+
+    component._change_slot_states(slots=slots, new_state=0)
+    assert component.nodes == [{'mem': 1024,
+                                'cores': [0, 0, 0, 0, 0, 0, 0, 0, 
+                                          0, 0, 0, 0, 0, 0, 0, 0],
+                                'uid': 2,
+                                'lfs': {'path': 'test', 'size': 100},
+                                'gpus':[0,0,0]}]
+
+
+    component.nodes = [{'uid': 2,
+                        'cores': [0, 0, 0, 0, 0, 0, 0, 0,
+                                  0, 0, 0, 0, 0, 0, 0, 0],
+                        'lfs': {'size': 100, 'path': 'test'},
+                        'mem': 1024,
+                        'gpus': [0, 0, 0]}]
+
+    slots = {"cores_per_node": 16,
+             "lfs_per_node": {"size": 0, "path": "/dev/null"},
+             "nodes": [{"lfs": {"path": "/dev/null", "size": 0},
+                        "core_map": [[0]],
+                        "name": "a",
+                        "gpu_map": [],
+                        "uid": 1,"mem": None}],
+             "lm_info": "INFO",
+             "gpus_per_node": 6,
+             }
+
+    with pytest.raises(RuntimeError):
+        component._change_slot_states(slots=slots, new_state=1)
+
+
+# ------------------------------------------------------------------------------
+#
+@mock.patch.object(AgentSchedulingComponent, '__init__', return_value=None)
+@mock.patch.object(AgentSchedulingComponent, '_handle_cuda', return_value=True)
+@mock.patch.object(AgentSchedulingComponent, 'schedule_unit',
+                   return_value={"cores_per_node": 16,
+                                 "lfs_per_node": {"size": 0, "path": "/dev/null"},
+                                 "nodes": [{"lfs": {"path": "/dev/null", "size": 0},
+                                            "core_map": [[0]],
+                                            "name": "a",
+                                            "gpu_map": None,
+                                            "uid": 1,"mem": None}],
+                                 "lm_info": "INFO",
+                                 "gpus_per_node": 6,
+                                 })
+@mock.patch.object(AgentSchedulingComponent, '_change_slot_states',
+                   return_value=True)
+def test_try_allocation(mocked_init, mocked_schedule_unit, mocked_handle_cuda,
```

Comment:
Same here: unless I misunderstand the purpose of the method, it is testing the most basic task.  Which is fine - but we should make it easy to add test cases.